### DISCUSSION TITLE
Inject default params in Zodios client

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#148] [Ideas]
Suppose I have a url like `https://endpoint.com/:id/transactions` and I will always pass the same `id`. Is it possible to inject this parameter on the client so I don't need to add it to the call everytime and make the typechecker happy?

**Comment by ecyrbe** [comment]
Is it known at compile time?

**Comment by ecyrbe** [comment]
If so here what you can do:
```ts
const ID = 'your id' as const;

// In your endpoint decl
path: `/${ID}/transactions`

// You still declare the generic one in another endpoint
path: '/:id/transactions'
```

**Comment by thelinuxlich** [comment]
Yeah, that can work. I still think it would be more flexible to have a way to pass standard fields for the zodios client, as it already allows to pass standard fields to AxiosConfig

**Comment by ecyrbe** [comment]
I'll try to think about it for use cases where it's only known at runtime (from env variable). 
For now i don't have any Idea how this would be possible and still make ts happy.

Moving to discussion in case someone has an Idea and want to share it.

**Reply by thelinuxlich** [reply]
Could be a object with the queries, headers, params, body fields, all optional, which would union with the object of the client API (like it's already done for headers & AxiosRequestHeaders

**Reply by ecyrbe** [reply]
Could you come up with an example, as i think i don't understand what you mean.

**Reply by thelinuxlich** [reply]
https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBvAUFKAzAlhAGwBMBGALigGdgAnXAOwHNMd9iAmC6uplvQogGYutBswC+6dKEhQA8mGC449AIYF4iFAAVVNJeoA8mgHxSAxiupQ4i5WoJyARgCsoFBUpXrNyNHzYhCgAibARg9HEoCytgKABbCBpGaBRDABUoCAAPYAh6IkooACUISxoiQ24xABooAFd6AGt6OAB3ehMTAAoWOFdyKHSavtdOKF19XCN0swBKCkykEyhu1CgAOi3+lxI6rY2d9ihxOZj6ax33WAQ-ROSINYCBQdDwkax+DhCnPQjxOq2LwOZwuOZAA

**Reply by thelinuxlich** [reply]
Never mind, I think I'm mixing runtime and compile-time things
